connect/transforms/src/main/java/org/apache/kafka/connect/transforms/InsertField.java

```diff
@@ -137,7 +137,7 @@ public R apply(R record) {
     }
 
     private boolean isTombstoneRecord(R record) {
-        return record.value() == null;
+        return operatingValue(record) == null;
```

Comment:
Can we please clean up the logic? This method no longer returns true if it's just a tombstone record; it also returns true if the key is null for `InsertField$Key`.

I'd suggest removing this method altogether and just changing the point where this method is called to simply be:
```
          if (operatingValue(record) == null) {
```

Reply:
Plus one for this; confusion around the term `tombstone` has already caused plenty of miscommunication around this issue and we should be careful to use it correctly.